property-scalaz/src/main/scala/testz/property.scala

```diff
@@ -45,34 +44,36 @@ object property {
     [F[_], I]
     (seed: (Array[Int], Int))
     (gen: Int => I)
-    (testGenerator: I => F[List[TestError]])
+    (testGenerator: I => F[TestResult])
     (numTestCases: Int)
-    (implicit F: Applicative[F]): F[List[TestError]] = {
+    (implicit F: Applicative[F]): F[TestResult] = {
     val rng = Well44497a.fromSeed(seed)
-    var acc: F[List[TestError]] = F.pure(Nil)
+    var acc: F[TestResult] = F.pure(Success)
     var i = 0
     while (i < numTestCases) {
-      acc = F.apply2(acc, testGenerator(gen(rng.nextInt)))(_ ++ _)
+      acc = F.apply2(acc, testGenerator(gen(rng.nextInt)))(TestResult.combine)
       i = i + 1
     }
-    acc
+    if (numTestCases == 0) F.pure(Failure.string("No test cases passed to `fromSeed`"))
+    else acc
   }
 
+  // repeatable tests, via the use of a constant seed.
   def repeatable
     [F[_], I]
     (gen: Int => I)
-    (testGenerator: I => F[List[TestError]])
+    (testGenerator: I => F[TestResult])
     (numTestCases: Int)
-    (implicit F: Applicative[F]): F[List[TestError]] =
+    (implicit F: Applicative[F]): F[TestResult] =
     fromSeed[F, I]((Array.fill[Int]((44497 + 31) / 32)(1010101010), 2))(gen)(testGenerator)(numTestCases)
 
   def nonrepeatable
     [F[_], I]
     (gen: Int => I)
-    (testGenerator: I => F[List[TestError]])
+    (testGenerator: I => F[TestResult])
     (numTestCases: Int)
     (fromTask: Task ~> F)
-    (implicit F: Monad[F]): F[List[TestError]] =
+    (implicit F: Monad[F]): F[TestResult] =
     for {
       seed <- fromTask(Task.delay(
         Array.fill[Array[Int]](10)(
```

Comment:
Does it actually run slow enough to produce different array elements? `System.nanoTime` would give a bit more entropy. But either way it won't be *significantly* better than a single `System.nanoTime`.

Reply:
The state space is two longs wide.

Reply:
It doesn't matter how large your state space is if you don't have enough entropy. E.g. say I fill the first 3 bits randomly and the rest with some deterministic function of those 3 bits, I'll have only 8 possible states.

Reply:
I'm just going to delete this seed-based code in a subsequent PR.